### ISSUE TITLE
useMeasure hook give wrong  values  : x,y,top,left are always 0

### ISSUE THREAD

**Issue by MarwenLabidi** [react-hookz/web#951]
**simple example :** https://codesandbox.io/s/trusting-bush-4nhxe3?file=/src/App.js
**What is the expected behavior?**
I expect the result of useMeasure  top left x and y to match the ones from the DOM.

**Comment by JoeDuncko** [comment]
@MarwenLabidi your `app.js` seems to be missing the useMeasure. This is all I see:

```
import "./styles.css";

export default function App() {
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>
    </div>
  );
}

```

**Comment by MarwenLabidi** [comment]
@JoeDuncko done , sorry for that

**Comment by JoeDuncko** [comment]
Got a chance to look at this. It looks like our [example](https://react-hookz.github.io/web/?path=/docs/sensor-usemeasure--example) also doesn't have an example where the top, left, x, and y show anything but 0. I also can't force it to do anything but be 0.

I haven't played with ResizeObserver enough to know if this is expected behavior or not. Will research and find out. Or someone can chime in :P .

**Comment by ArttuOll** [comment]
I can also take a look at this today. Will report here, if I find out anything.

**Comment by ArttuOll** [comment]
When we use this hook, we are giving a `ref` to the element we want to measure. We can access this element's bounding rect by calling `.getBoundingClientRect()` on it. This rect contains the measures we are interested in here.

When I access the example's measured element using the Chrome devtools, I see very different values for `x` and `y` than 0, which seems to confirm, that there indeed is a bug here. I also get the differing measurements when I log the measurements of the `elementRef` on line 19 of the hook. The `x` and `y` properties are supposed to be relative to the viewport, which they clearly are not at the moment.

Continuing to investigate.

**Comment by ArttuOll** [comment]
>  I also can't force it to do anything but be 0.

I found that the coordinates can be changed by adding padding to the measured element.

I explain the reason in my draft PR #954 : 
> The contentRect property's x and y coordinates, as well as the top and left properties, are defined by the element's top and left padding, as explained in the [spec](https://w3c.github.io/csswg-drafts/resize-observer-1/#content-rect). This does not appear to be the case for the rect returned by `getBoundingClientRect`.

**Comment by lensbart** [comment]
The same issue exists for `react-use`. [This PR](https://github.com/streamich/react-use/pull/2031) might be helpful.

**Comment by ArttuOll** [comment]
Oh, nice catch! I wish I had taken a look at `react-use` when debugging this... Would have saved hours of my life 😆

**Comment by ArttuOll** [comment]
Closed with #954 .
